tests/test_flows.py

```diff
@@ -401,18 +401,86 @@ def test_channel_selection(self):
         self.check_log(self.repeater1, "agent_wlan0", "CHANNEL_PREFERENCE_QUERY_MESSAGE")
         self.check_log(self.repeater1, "agent_wlan2", "CHANNEL_PREFERENCE_QUERY_MESSAGE")
 
-        self.debug("Send channel selection request")
-        self.gateway_ucc.dev_send_1905(self.mac_repeater1, 0x8006)
-        time.sleep(1)
-        self.debug("Confirming channel selection request has been received on agent")
+        # Send empty channel selection request
+        self.debug("Send empty channel selection request")
+        self.gateway_ucc.dev_send_1905(self.mac_repeater1,0x8006)
+
+        self.debug("Confirming empty channel selection request has been received on agent")
         self.check_log(self.repeater1, "agent_wlan0", "CHANNEL_SELECTION_REQUEST_MESSAGE")
         self.check_log(self.repeater1, "agent_wlan2", "CHANNEL_SELECTION_REQUEST_MESSAGE")
 
-        #self.debug("Confirming 1905.1 Ack Message request was received on agent")
-        # TODO: When creating handler for the ACK message on the agent, replace lookup of this string
-        # TODO: currently controller sends empty channel selection request, so no switch is performed
-        # self.check_log(self.repeater1, "agent_wlan0", "ACK_MESSAGE")
-        # self.check_log(self.repeater1, "agent_wlan2", "ACK_MESSAGE")
+        self.debug("Confirming OPERATING_CHANNEL_REPORT_MESSAGE message has been received on controller")
+        self.check_log(self.gateway, "controller", "OPERATING_CHANNEL_REPORT_MESSAGE")
+
+        # Send channel selection request with chnging tx_power_limit 
+        # Payload for transmit power (tp)
+        tp20dBm = "0x14"
+        tp21dBm = "0x15"
+
+        for payload_transmit_power in (tp20dBm,tp21dBm):
+            transmit_power = int(payload_transmit_power,16)
+            self.debug("Send empty channel selection request with changing tx_power_limit")
+            self.gateway_ucc.dev_send_1905(
+                self.mac_repeater1,
+                0x8006,
+                tlv(0x8D,0x0007, f'{self.mac_repeater1_wlan0} {payload_transmit_power}'),
+                tlv(0x8D,0x0007, f'{self.mac_repeater1_wlan2} {payload_transmit_power}')
```

Comment:
Unfortunately, @tomereli and friends are stuck with Python 3.5 or worse, so no f-strings...

```suggestion
                tlv(0x8D, 0x0007, '{} {:02x}'.format(self.mac_repeater1_wlan0, payload_transmit_power)),
                tlv(0x8D, 0x0007, '{} {:02x}'.format(self.mac_repeater2_wlan0, payload_transmit_power)),
```

(and with f-string it could have been `tlv(0x8D,0x0007, f'{self.mac_repeater1_wlan0} {payload_transmit_power:02x}

Reply:
This solution does not work. It should be like:

`tlv(0x8D, 0x0007, '{} {}'.format(self.mac_repeater1_wlan0, hex(payload_transmit_power))),`

Reply:
What does not work about it?

`hex` is not good, because it doesn't include a leading zero for values < 16.

Oh, now I see, I forgot the 0x bit, is that it?
```suggestion
                tlv(0x8D, 0x0007, '{} 0x{:02x}'.format(self.mac_repeater1_wlan0, payload_transmit_power)),
                tlv(0x8D, 0x0007, '{} 0x{:02x}'.format(self.mac_repeater1_wlan0, payload_transmit_power))
```

Reply:
Sure, but we should pay attention on `#define TX_POWER_MIN 19`